### PR TITLE
Group disabled policies together when sorting by status

### PR DESCRIPTION
**Pull request by austinbyers**
## Background

When sorting compliance policies by status, they do not appear to be grouped correctly:

Screen Shot 2020-08-26 at 1 04 21 PM

The "Disabled" status shows up among the "Pass" status - this is because even disabled policies are still evaluated, we just don't yet show its underlying status to the user (we should eventually).

Until then, the backend will first sort by enabled/disabled status so that the column sorts as you would expect

Closes: #1179 

## Changes

- When sorting policies by status, sort first by enabled/disabled (to group all disabled policies together at the end)
- Add some clarifying comments

## Testing

- After applying the change and sorting on the column:

Screen Shot 2020-08-26 at 1 10 57 PM

- `PKG=./internal/core/analysis_api/main dev -- mage test:integration`

### CODE REVIEW

**Comment by austinbyers** [comment]
If this page were redesigned (e.g. enabled/disabled made into a separate column), then we can re-tweak the sorting as needed. This was just the fastest simplest solution which makes it work much better for the user right now

File: internal/core/analysis_api/handlers/list.go [hunk 1]
```diff
@@ -426,13 +443,13 @@ func sortByEnabled(policies []*models.PolicySummary, ascending bool) {
 	sort.Slice(policies, func(i, j int) bool {
 		left, right := policies[i], policies[j]
 		if left.Enabled && !right.Enabled {
-			// ascending: true > false
-			return !ascending
+			// when ascending (default): enabled < disabled
+			return ascending
```
Comment (austinbyers) [inline internal/core/analysis_api/handlers/list.go:447]: Flipped so that enabled comes before disabled by default

**Comment by 3nvi** [comment]
Thanks for that @austinbyers . I  didn't communicate the  fact that I was already in discussion with  our designer to make sure that we don't group those 2 statuses in 1 UI component.

The result can be seen here https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/master/commits/d0570e02d9fb58bb4e79266e5e7ff6e3978973a8/files/323d0ca1-81a8-40e7-83ee-0a05b870bb4a/layers/4B95AE99-746B-43F4-8FC0-CF251074C34F

With the new design this change won't be needed, although this design won't land until 1.9 Until then, we can definitely keep this work, but,  if the operations performed in this PR are costly, we'll have to remember to revert it

**Comment by austinbyers** [comment]
> Until then, we can definitely keep this work, but, if the operations performed in this PR are costly, we'll have to remember to revert it

Sounds good, I'll merge for now. It's not costly to do the extra sort, but it won't make sense to keep once it's a separate column, so I'll add a TODO so we remember to remove it

**Pull request merged**